src/src/rx_main.cpp

```diff
@@ -878,4 +972,150 @@ void loop()
+    // and we're not already in binding mode, enter binding
+    if (!config.GetIsBound() && !InBindingMode)
+    {
+        Serial.println("RX has not been bound, enter binding mode...");
+        EnterBindingMode();
+    }
+
+    // If the power on counter is >=3, enter binding and clear counter
+#ifndef MY_UID
+    if (config.GetPowerOnCounter() >= 3)
+    {
+        config.SetPowerOnCounter(0);
+        config.Commit();
+        
+        Serial.println("Power on counter >=3, enter binding mode...");
+        EnterBindingMode();
+    }
+#endif
+    // Update the LED while in binding mode
+    if (InBindingMode)
+    {
+        if (millis() > bindLedFlashInterval)
+        {
+            if (LEDPulseCounter == 0)
+            {
+                LED = true;
+            }
+            else if (LEDPulseCounter == 4)
+            {
+                LED = false;
+            }
+            else
+            {
+                LED = !LED;
+            }
+
+            if (LEDPulseCounter < 4)
+            {
+                bindLedFlashInterval = millis() + BIND_LED_FLASH_INTERVAL_SHORT;
+            }
+            else
+            {
+                bindLedFlashInterval = millis() + BIND_LED_FLASH_INTERVAL_LONG;
+                LEDPulseCounter = 0;
+            }
+            
+            
+            #ifdef GPIO_PIN_LED
+            digitalWrite(GPIO_PIN_LED, LED);
+            #endif
+            
+            LEDPulseCounter++;
+        }
+    }
+}
+
+void EnterBindingMode()
+{
+    if ((connectionState == connected) || InBindingMode || webUpdateMode) {
+        // Don't enter binding if:
+        // - we're already connected
+        // - we're already binding
+        // - we're in web update mode
+        Serial.println("Cannot enter binding mode!");
+        return;
+    }
+
+    // Set UID to special binding values
+    UID[0] = BindingUID[0];
+    UID[1] = BindingUID[1];
+    UID[2] = BindingUID[2];
+    UID[3] = BindingUID[3];
+    UID[4] = BindingUID[4];
+    UID[5] = BindingUID[5];
+
+    CRCCaesarCipher = UID[4];
+    DeviceAddr = UID[5] & 0b111111;
+
+    // Start attempting to bind
+    // Lock the RF rate and freq while binding
+    SetRFLinkRate(RATE_DEFAULT);
+    Radio.SetFrequency(GetInitialFreq());
+
+    InBindingMode = true;
+
+    Serial.print("Entered binding mode at freq = ");
+    Serial.println(Radio.currFreq);
+}
+
+void ExitBindingMode()
+{
+    if (!InBindingMode) {
+        // Not in binding mode
+        Serial.println("Cannot exit binding mode, not in binding mode!");
+        return;
+    }
+
+    InBindingMode = false;
+
+    LostConnection();
+
+    // Sets params to allow sync after binding and not instantly change RF mode
+    RFmodeCycleDivisor = 1;
+    RFmodeLastCycled = millis();
+
+    Serial.print("Exit binding mode at freq = ");
+    Serial.println(Radio.currFreq);
+}
+
+void OnELRSBindMSP(mspPacket_t *packet)
+{
+    UID[2] = packet->readByte();
```

Comment:
May I ask why UID is 6 characters, while only last 4 are transferred from TX to RX and used for calculations and checks? UID[0] and UID[1] are never read, only set.
I think refactoring it to 4 digit UID might be beneficial to save some EEPROM space and speed up read/write. What do you think?

Reply:
Long term it will be useful to have all 6.  I dont think eeprom space or speed is an issue ATM.